src/main/scala/dpla/ingestion3/MappingEntry.scala

```diff
@@ -71,36 +78,34 @@ object MappingEntry {
     }
 
     // Run the mapping over the Dataframe
-    val mappingResults = harvestedRecords.select("document").map(
-      record => {
-        extractorClass.getConstructor(classOf[String])
-          .newInstance(record.getAs[String]("document"))
-          .build
-      }
-    )
-
-    // TODO there is probably a much cleaner/better way of writing this.
-    val mappingSuccess = mappingResults
-      .filter(r => r.isInstanceOf[DplaMapData])
-      .map(r2 => r2.asInstanceOf[DplaMapData])
-
-    // TODO ditto ^^ This converts the Dataset to an Array[String], This is an easy way out for now.
-    val failures = mappingResults
-      .filter(r => r.isInstanceOf[DplaMapError])
-      .map(r2 => r2.asInstanceOf[DplaMapError]).map(f => f.errorMessage).collect()
+    val documents = harvestedRecords.select("document").as[String]
```

Comment:
I think that in the line above, `documents` is a `Dataset[String]` whereas `harvestedRecords` is a `DataFrame`?  Is that correct?  Maybe would be nice to specify when declaring the `val`'s.

Reply:
@AudreyAltman concur